core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -47,49 +51,52 @@ impl TeeRequestProcessor {
     ) -> Result<Option<Json<TeeProofGenerationDataResponse>>, RequestProcessorError> {
         tracing::info!("Received request for proof generation data: {:?}", request);
 
-        let mut min_batch_number = self.config.tee_config.first_tee_processed_batch;
-        let mut missing_range: Option<(L1BatchNumber, L1BatchNumber)> = None;
+        let batch_ignored_timeout = ChronoDuration::days(10);
+        let min_batch_number = self.config.tee_config.first_tee_processed_batch;
 
-        let result = loop {
-            let Some(l1_batch_number) = self
+        loop {
+            let Some(locked_batch) = self
                 .lock_batch_for_proving(request.tee_type, min_batch_number)
                 .await?
             else {
-                // No job available
-                return Ok(None);
+                return Ok(None); // no job available
             };
+            let batch_number = locked_batch.l1_batch_number;
 
             match self
-                .tee_verifier_input_for_existing_batch(l1_batch_number)
+                .tee_verifier_input_for_existing_batch(batch_number)
                 .await
             {
                 Ok(input) => {
                     break Ok(Some(Json(TeeProofGenerationDataResponse(Box::new(input)))));
                 }
                 Err(RequestProcessorError::ObjectStore(ObjectStoreError::KeyNotFound(_))) => {
-                    missing_range = match missing_range {
-                        Some((start, _)) => Some((start, l1_batch_number)),
-                        None => Some((l1_batch_number, l1_batch_number)),
+                    let duration = Utc::now().signed_duration_since(locked_batch.created_at);
+                    let status = if duration > batch_ignored_timeout {
+                        TeeProofGenerationJobStatus::PermanentlyIgnored
+                    } else {
+                        TeeProofGenerationJobStatus::Failed
                     };
-                    self.unlock_batch(l1_batch_number, request.tee_type).await?;
-                    min_batch_number = l1_batch_number + 1;
+                    self.unlock_batch(batch_number, request.tee_type, status)
```

Comment:
I feel we've already had this conversation: This looks like backend driven by frontend anti-pattern; batches are only unlocked in response to client requests. I'd imagine that batches should be unlocked on a timeout (currently hard-coded as 10 days) with `PermanentlyIgnored` status, right? Or is there no harm if a batch is unlocked later?

Reply:
There is also a `unlock` part in the SQL query for new jobs ... 
```
OR ( tee.status = `picked_by_prover` AND tee.prover_taken_at < NOW() - processing_timeout::INTERVAL )
```

Reply:
> (...) Or is there no harm if a batch is unlocked later?

There is no harm I can think of. Also, please consider the point @haraldh mentioned above (and that I elaborated on further [here](https://github.com/matter-labs/zksync-era/pull/2980/files/9188f8a3107f9a46d5b74c99efe7a1db5ccda56d#r1847009731)): batch status can transition from `failed` to `picked_by_prover`, and this transition is also driven by frontend (since [`lock_batch_for_proving`](https://github.com/matter-labs/zksync-era/blob/9188f8a3107f9a46d5b74c99efe7a1db5ccda56d/core/lib/dal/src/tee_proof_generation_dal.rs#L53) is ultimately triggered by a user request).

Reply:
> batch status can transition from failed to picked_by_prover,

This one is expected, because picking batches is *expected* to be driven by the client requesting proof inputs (unless I'm horribly misunderstanding the entire workflow). It is unlocking that looks suboptimal because *conceptually*, rows in `tee_proof_generation_details` should be unlocked w/ `PermanentlyIgnored` state on a certain timeout after their creation (of course only if they don't reach another terminal state before that). If this is indeed the ideal workflow, it can be implemented as a background task rather than being driven by the frontend. Can be implemented separately, of course.

Reply:
> (...) If this is indeed the ideal workflow, it can be implemented as a background task rather than being driven by the frontend. (...)

Do we already have a similar background task implemented somewhere? I would rather not reinvent the wheel if we have some building blocks for creating this kind of background service that I can reuse.

> (...) Can be implemented separately, of course.

Alright, I’ll probably handle it in a separate PR then.